### PR TITLE
Add profile recording e2e test

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
We now add an OCI hook based recording test to our e2e test suite.

#### Which issue(s) this PR fixes:

None
#### Does this PR have test?
Yes

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
This works now and is ready for review.

**Comment by saschagrunert** [comment]
Or not
/hold

File: test/tc_profilerecordings_test.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	e.Nil(err)
+	e.kubectl("create", "-f", testPodFile.Name())
+
+	e.logf("Waiting for test pod to be initialized")
+	e.waitFor("condition=initialized", "pod", "recording")
+
+	output := e.kubectl("get", "pod", "recording")
+	for strings.Contains(output, "ContainerCreating") {
+		output = e.kubectl("get", "pod", "recording")
+	}
```
Comment (JAORMX) [inline test/tc_profilerecordings_test.go:64]: why not wait for the pod to be ready?

Comment (saschagrunert) [reply test/tc_profilerecordings_test.go]: Yeah this makes more sense, I changed the test logic which should also simplify the test.

File: test/tc_profilerecordings_test.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	const testPod = `
+apiVersion: v1
+kind: Pod
+metadata:
+  name: recording
+  labels:
+    app: alpine
+spec:
+  containers:
+  - image: docker.io/nginx:alpine
```
Comment (JAORMX) [inline test/tc_profilerecordings_test.go:43]: can you push this image to the SPO's quay.io namespace?

Comment (saschagrunert) [reply test/tc_profilerecordings_test.go]: I'll use the 1.19.1 nxing image from quay :)

**Pull request merged**